codec/decoder/core/loongarch/mb_aux_lsx.c

```diff
@@ -0,0 +1,292 @@
+/*!
+ * \copy
+ *     Copyright (c)  2022, Cisco Systems
+ *     All rights reserved.
+ *
+ *     Redistribution and use in source and binary forms, with or without
+ *     modification, are permitted provided that the following conditions
+ *     are met:
+ *
+ *        * Redistributions of source code must retain the above copyright
+ *          notice, this list of conditions and the following disclaimer.
+ *
+ *        * Redistributions in binary form must reproduce the above copyright
+ *          notice, this list of conditions and the following disclaimer in
+ *          the documentation and/or other materials provided with the
+ *          distribution.
+ *
+ *     THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *     "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *     LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *     FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *     COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *     INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *     BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *     LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *     CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *     LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *     ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *     POSSIBILITY OF SUCH DAMAGE.
+ *
+ *
+ * \file    mb_aux_lsx.c
+ *
+ * \brief   Loongson optimization
+ *
+ * \date    7/3/2022 Created
+ *
+ *************************************************************************************
+ */
+
+#include <stdint.h>
+#include "loongson_intrinsics.h"
+
+#define DO(in0, in1, in2, in3, in4, in5, in6, in7, \
+           ou0, ou1, ou2, ou3, ou4, ou5, ou6, ou7) \
+do {                                               \
+  __m128i t0, t1, t2, t3, a0, a1, a2, a3;          \
+  __m128i b0, b1, b2, b3, b4, b5, b6, b7;          \
```

Comment:
you can use 'out*' instead of applying so many variable.

Reply:
When calling the macro 'DO', the src0 ... src7 are passed to the input and output parameters, so using 'out*' instead of defing local variables needs to define extra variables like out0 ... out7 and do some variables adjustment. That seems to benefit nothing. Thanks !